### PR TITLE
introduce SalesChannelRequestContextResolverInterface

### PR DESCRIPTION
**Pull request by raphael-homann**
### 1. Why is this change necessary?
I need to decorate SalesChannelRequestContextResolver, but SalesChannelProxyController requires SalesChannelRequestContextResolver, not an interface.

### 2. What does this change do, exactly?
a new interface has been added and used for service injection.

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Framework/Api/Controller/SalesChannelProxyController.php [hunk 1]
```diff
@@ -77,7 +77,7 @@ public function __construct(
         private readonly EntityRepository $salesChannelRepository,
         protected DataValidator $validator,
         protected SalesChannelContextPersister $contextPersister,
-        private readonly SalesChannelRequestContextResolver $requestContextResolver,
+        private readonly SalesChannelRequestContextResolverInterface $requestContextResolver,
```
Comment (shyim) [inline src/Core/Framework/Api/Controller/SalesChannelProxyController.php:80]: We should maybe change this to `RequestContextResolverInterface` and then you can decorate it using the normal symfony interface 🤔

Comment (raphael-homann) [reply src/Core/Framework/Api/Controller/SalesChannelProxyController.php:80]: That was my first implementation, but I introduced SalesChannelRequestContextResolverInterface to be sure it is a SalesChannelRequest....
but it's more simple, and works fine for me too.

Comment (keulinho) [reply src/Core/Framework/Api/Controller/SalesChannelProxyController.php:80]: Please change this to the symfony type hint, there is no need to introduce a new interface for that needs to be maintained etc if there is no added benefit for that

**Comment by keulinho** [comment]
And also please sign the [Contributor License Agreement](https://cla-assistant.io/shopware/shopware?pullRequest=3841), otherwise we can't import your PR.

**Comment by keulinho** [comment]
Hey @raphael-homann i will close this due to inactivity. 

Feel free to reopen the MR when you can address the CR comments.

**Pull request closed without merging**